pytorch_lightning/accelerators/base_backend.py

```diff
@@ -165,3 +165,9 @@ def on_train_epoch_end(self):
 
     def early_stopping_should_stop(self, pl_module):
         return self.trainer.should_stop
+
+    def setup_optimizers(self, model):
```

Comment:
+1 for putting this in the base class. I like it. 
Have you thought about moving the "if self.testing" logic here as well?
It looks like it is repeated everywhere, so it could simply be handled here. what do you think?

Reply:
yep, good point.